socs/agents/smurf_crate_monitor/agent.py

```diff
@@ -271,6 +271,78 @@ def _stop_acq(self, session, params=None):
         else:
             return False, 'acq is not currently running'
 
+    @ocs_agent.param('slot', type=int, choices=[1, 2, 3, 4, 5, 6, 7], default=None)
+    def deactivate_slot(self, session, params=None):
+        """
+        deactivate_slot()
+
+        **Task** - Deactivates a specific slot in the crate.
+        This is used to ensure that the boards are not active
+        when the boards are being worked on or air cooling is blocked.
+
+        Args:
+            slot (int):
+                Slot number to deactivate. Allowed values are 1-7.
+        """
+        slot = params.get('slot')
```

Comment:
Once you've decorated with param, you don't have to use .get anymore.  `params['slot']` is guaranteed to be None or one of those integers.

If you remove the "default=None" line, from the decorator, then None will be forbidden -- so then you don't have to handle that as an error, yourself -- caller will get an automatic informative error.